### PR TITLE
Have the secret key API endpoint return a file like the public one does

### PR DESCRIPTION
**Pull request by raskchanky**
tenor-195437959

Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-depot/src/server.rs [hunk 1]
```diff
@@ -639,10 +639,16 @@ fn download_latest_origin_secret_key(req: &mut Request) -> IronResult<Response>
...
-    }
+    let key = match route_message::<OriginSecretKeyGet, OriginSecretKey>(req, &request) {
+        Ok(key) => key,
+        Err(err) => {
+            error!("Can't retrieve secret key file: {}", err);
+            return Ok(render_net_error(&err));
+        }
+    };
+
+    let xfilename = format!("{}-{}.sig.key", key.get_name(), key.get_revision());
```
Comment (christophermaier) [inline components/builder-depot/src/server.rs]: Is this key formatting logic present anywhere else in the codebase? It'd be nice to reuse if possible.

Comment (raskchanky) [reply components/builder-depot/src/server.rs]: I grepped through and the places I found that were outputting keys were formatting things manually, as here.  It'd be nice to consolidate all that into one place, but I think that's out of scope for this PR.

**Comment by mwrock** [comment]
LGTM!

**Comment by raskchanky** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/281061293).

**Comment by fnichol** [comment]
Won’t this affect the logic in builder-worker? That should be only part of the system (currently) that consumes this endpoint.

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/281061293).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by raskchanky** [comment]
@fnichol I thought @chefsalim refactored builder-worker to not use builder-http-client?  Ideally, we want all our services communicating with each other over zmq, not HTTP.

**Comment by fnichol** [comment]
Oh sweet, if that's the case, even better--that's where we're headed. I'm just looking out for the worker at the moment 😄

**Comment by reset** [comment]
@raskchanky @fnichol actually that change was backed out because the worker is supposed to communicate with a gateway to re-insert artifacts into the system. It's true that we want all services communicating behind the Router to use ZMQ but for the Worker, we have it submit data it generated through the public gateway as an extra security measure

**Comment by raskchanky** [comment]
@fnichol @reset In that case, it's worth taking a look at the worker and making sure it still works.  I'll make an issue.

**Pull request merged**